godal.go

```diff
@@ -3733,6 +3762,55 @@ func BuildVRT(dstVRTName string, sourceDatasets []string, switches []string, opt
 	return &Dataset{majorObject{C.GDALMajorObjectH(hndl)}}, nil
 }
 
+// GridCreate, creates a grid from scattered data, given provided gridding parameters as a string (pszAlgorithm)
+// and the arguments required for `godalGridCreate()` (binding for GDALGridCreate)
+//
+// NOTE: For valid gridding algorithm strings see: https://gdal.org/programs/gdal_grid.html#interpolation-algorithms
+func GridCreate(pszAlgorithm string,
+	numCoords int,
```

Comment:
let's not add `numCoords` to our public API, as it can be extracted from `len(xCoords)`. Also add a check that `len(xCoords)==len(yCoords)==len(zCoords)` (I'm not familiar with the API, maybe len(zCoords) is allowed to be 0)